pipert/core/pipeline_manager.py

```diff
@@ -0,0 +1,441 @@
+                str(error)
+            )
+
+    @component_name_existence_error(need_to_be_exist=True)
+    def remove_routine_from_component(self, component_name, routine_name):
+        if self._does_component_running(self.components[component_name]):
+            return self._create_response(
+                False,
+                "You can't remove a routine while your component is running"
+            )
+        self.components[component_name].remove_routine(routine_name)
+        return self._create_response(
+            True,
+            f"Removed routines with the name {routine_name} from the component"
+        )
+
+    @component_name_existence_error(need_to_be_exist=True)
+    def create_queue_to_component(self, component_name,
+                                  queue_name, queue_size=1):
+        if self.components[component_name].does_queue_exist(queue_name):
+            return self._create_response(
+                False,
+                f"Queue named {queue_name} already exist"
+            )
+
+        self.components[component_name].create_queue(queue_name=queue_name,
+                                                     queue_size=queue_size)
+        return self._create_response(
+            True,
+            f"The Queue {queue_name} has been created"
+        )
+
+    @component_name_existence_error(need_to_be_exist=True)
+    def remove_queue_from_component(self, component_name, queue_name):
+        if not self.components[component_name].does_queue_exist(queue_name):
+            return self._create_response(
+                False,
+                f"Queue named {queue_name} doesn't exist"
+            )
+
+        if self.components[component_name]. \
+                does_routines_use_queue(queue_name):
+            return self._create_response(
+                False,
+                f"Can't remove a queue that is being used by routines"
+            )
+
+        self.components[component_name].delete_queue(queue_name=queue_name)
+        return self._create_response(
+            True,
+            f"The Queue {queue_name} has been removed"
+        )
+
+    @component_name_existence_error(need_to_be_exist=True)
+    def run_component(self, component_name):
+        if self._does_component_running(self.components[component_name]):
+            return self._create_response(
+                False,
+                f"The component {component_name} already running"
+            )
+        else:
+            self.components[component_name].run()
+            return self._create_response(
+                True,
+                f"The component {component_name} is now running"
+            )
+
+    @component_name_existence_error(need_to_be_exist=True)
+    def stop_component(self, component_name):
+        if not self._does_component_running(self.components[component_name]):
+            return self._create_response(
+                False,
+                f"The component {component_name} is not running running"
```

Comment:
double running

Reply:
I don't understand the comment, can you be a little bit more specific